src/main/java/org/jenkinsci/plugins/githubreleaseperformer/GitHubReleasePerformer.java

```diff
@@ -0,0 +1,225 @@
+package org.jenkinsci.plugins.githubreleaseperformer;
+
+import hudson.Extension;
+import hudson.Launcher;
+import hudson.Util;
+import hudson.model.AbstractBuild;
+import hudson.model.AbstractProject;
+import hudson.model.BuildListener;
+import hudson.tasks.BuildStepDescriptor;
+import hudson.tasks.Builder;
+import net.sf.json.JSONObject;
+import org.apache.commons.io.IOUtils;
+import org.jenkinsci.plugins.tokenmacro.TokenMacro;
+import org.kohsuke.github.GHContent;
+import org.kohsuke.github.GHReleaseBuilder;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.github.GitHub;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.StaplerRequest;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.StringWriter;
+
+public class GitHubReleasePerformer extends Builder {
+
+	private final String apiUrl;
+	private final String tag;
+	private final String branch;
+	private final String releaseNotesFile;
+	private final String user;
+	private final String password;
+	private final String owner;
+	private final String repository;
+
+	private GitHub github;
+	private GHRepository ghRepository;
+
+	private static final String RELEASE_NOTES_TEMPLATE =
```

Comment:
What do we do about major releases? In some cases it's not just New Features/Improvements/Bug fixes, i.e. https://github.com/Ontotext-AD/concept-extraction-pub/releases/tag/euf-2.0.0

We can either:
1. say ok, these are added manually for major releases
2. try to have a more flexible template

I'm happy with the first option, it just has to be made clear to the team when they need to edit the notes manually.